circuitpython_build_tools/build.py

```diff
@@ -266,17 +271,20 @@ def library(library_path, output_directory, package_folder_prefix,
                 ])
                 if mpy_success != 0:
                     raise RuntimeError("mpy-cross failed on", full_path)
-            else:
-                shutil.copyfile(temp_file.name, output_file)
+            temp_filename = temp_file.name
+        shutil.copyfile(temp_filename, output_file)
+        os.remove(temp_filename)
 
     for filename in package_files:
         full_path = os.path.join(library_path, filename)
-        with tempfile.NamedTemporaryFile() as temp_file:
+        temp_filename = ""
+        output_file = ""
```

Comment:
not sure why these need to be assigned empty values here, but it's harmless.

Reply:
Scoping reasons. Previously, `shutil.copyfile()` had been conditionally using them inside the `with` block.

I went with empty strings as init because this library could use type hinting (maybe sometime in the future).